examples/website/image-tile/app.js

```diff
@@ -14,121 +15,79 @@ const INITIAL_VIEW_STATE = {
 const ROOT_URL =
   'https://raw.githubusercontent.com/visgl/deck.gl-data/master/website/image-tiles/moon.image';
 
-export default class App extends PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      height: null,
-      width: null,
-      tileSize: null
-    };
-    this._onHover = this._onHover.bind(this);
-    this._renderTooltip = this._renderTooltip.bind(this);
-    this.inTileBounds = this.inTileBounds.bind(this);
-    this.cutOffImageBounds = this.cutOffImageBounds.bind(this);
-    this.fetchDataFromDZI(`${ROOT_URL}/moon.image.dzi`);
-  }
-
-  _onHover({x, y, sourceLayer, tile}) {
-    this.setState({x, y, hoveredObject: {sourceLayer, tile}});
-  }
-
-  _renderTooltip() {
-    const {x, y, hoveredObject} = this.state;
-    const {sourceLayer, tile} = hoveredObject || {};
-    return (
-      sourceLayer &&
-      tile && (
-        <div className="tooltip" style={{left: x, top: y}}>
-          tile: x: {tile.x}, y: {tile.y}, z: {tile.z}
-        </div>
-      )
-    );
-  }
-
-  inTileBounds({x, y, z}) {
-    const xInBounds = x < Math.ceil(this.state.width / (this.state.tileSize * 2 ** z)) && x >= 0;
-    const yInBounds = y < Math.ceil(this.state.height / (this.state.tileSize * 2 ** z)) && y >= 0;
-    return xInBounds && yInBounds;
-  }
+function getTooltip({tile}) {
+  return tile
+    ? {
+        className: 'tooltip',
+        text: `tile: x: ${tile.x}, y: ${tile.y}, z: ${tile.z}`
+      }
+    : null;
+}
 
-  cutOffImageBounds({left, bottom, right, top}) {
-    return {
-      left: Math.max(0, left),
-      bottom: Math.max(0, Math.min(this.state.height, bottom)),
-      right: Math.max(0, Math.min(this.state.width, right)),
-      top: Math.max(0, top)
-    };
-  }
+export default function App({autoHighlight = true}) {
+  const [dimensions, setDimensions] = useState(null);
 
-  fetchDataFromDZI(dziSource) {
-    return (
-      fetch(dziSource) // eslint-disable-line no-undef
-        .then(response => response.text())
-        // eslint-disable-next-line no-undef
-        .then(str => new window.DOMParser().parseFromString(str, 'text/xml'))
-        .then(dziXML => {
-          if (Number(dziXML.getElementsByTagName('Image')[0].attributes.Overlap.value) !== 0) {
-            // eslint-disable-next-line no-undef, no-console
-            console.warn('Overlap paramter is nonzero and should be 0');
-          }
-          this.setState({
-            height: Number(dziXML.getElementsByTagName('Size')[0].attributes.Height.value),
-            width: Number(dziXML.getElementsByTagName('Size')[0].attributes.Width.value),
-            tileSize: Number(dziXML.getElementsByTagName('Image')[0].attributes.TileSize.value)
-          });
-        })
-    );
-  }
+  useEffect(() => {
```

Comment:
Nit: async await would clean this up quite a bit. Out of scope for this PR, but since this is about modernizing code...